torch/nn/_functions/dropout.py

```diff
@@ -24,21 +25,28 @@ def forward(self, input):
         else:
             output = input.clone()
 
-        if self.p > 0 and self.train:
-            self.noise = self._make_noise(input)
-            self.noise.bernoulli_(1 - self.p).div_(1 - self.p)
-            if self.p == 1:
-                self.noise.fill_(0)
-            self.noise = self.noise.expand_as(input)
-            output.mul_(self.noise)
+        if self.p > 0:
+            if self.train:
+                self.noise = self._make_noise(input)
+                self.noise.bernoulli_(1 - self.p)
+                if self.scale_train:
+                    self.noise.div_(1 - self.p)
+                if self.p == 1:
```

Comment:
can you move this up and make it (right now we unnecessarily call `bernoulli_` when `p == 1`)
```python
if self.p == 1:
    self.noise.fill_(0)
else:
    self.noise.bernoulli_(1 - self.p)
    if self.scale_train:
        self.noise.div_(1 - self.p)
```